sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/CosmosEncryptionAsyncContainer.java

```diff
@@ -1087,6 +1096,131 @@ ItemDeserializer getItemDeserializer() {
+                        cosmosItemOperation.getPartitionKeyValue(),
+                        ((ItemBulkOperation<JsonNode, TContext>) cosmosItemOperation).getRequestOptions(),
+                        encryptedItem,
+                        cosmosItemOperation.getContext()
+                    ));
+            } else {
+                cosmosItemOperationMono = Mono.just(
+                    new ItemBulkOperation<>(
+                        cosmosItemOperation.getOperationType(),
+                        cosmosItemOperation.getId(),
+                        cosmosItemOperation.getPartitionKeyValue(),
+                        ((ItemBulkOperation<JsonNode, TContext>) cosmosItemOperation).getRequestOptions(),
+                        null,
+                        cosmosItemOperation.getContext()
+                    )
+                );
+            }
+            return cosmosItemOperationMono;
+        });
+
+        final CosmosBulkExecutionOptions cosmosBulkExecutionOptions = bulkOptions;
+        Mono<Flux<CosmosItemOperation>> then = operationFlux.then(Mono.defer(() -> {
+            setRequestHeaders(cosmosBulkExecutionOptions);
+            return Mono.just(operationFlux);
+        }));
+
+        return then.flatMapMany(cosmosItemOperationFlux -> {
+            return executeBulkOperationsHelper(operationFlux, cosmosBulkExecutionOptions, false);
+        });
+    }
+
+    private <TContext> Flux<CosmosBulkOperationResponse<TContext>> executeBulkOperationsHelper(Flux<CosmosItemOperation> operations,
+                                                                                               CosmosBulkExecutionOptions bulkOptions,
+                                                                                               boolean isRetry) {
+        return this.container.executeBulkOperations(operations, bulkOptions).flatMap(cosmosBulkOperationResponse -> {
+
+            CosmosBulkItemResponse cosmosBulkItemResponse = cosmosBulkOperationResponse.getResponse();
+            ObjectNode objectNode = this.cosmosBulkItemResponseAccessor.getResourceObject(cosmosBulkItemResponse);
+
+            if(objectNode != null) {
+                Mono<JsonNode> jsonNodeMono = encryptionProcessor.decryptJsonNode(objectNode).flatMap(jsonNode -> {
+                    this.cosmosBulkItemResponseAccessor.setResourceObject(cosmosBulkItemResponse,
+                        (ObjectNode) jsonNode);
+                    int  x =1;
+//                    this.cosmosBulkOperationResponseAccessor.setResponse(cosmosBulkOperationResponse,
+//                        cosmosBulkItemResponse);
+                    return Mono.just(jsonNode);
+                });
+                return jsonNodeMono.flux().flatMap(jsonNode -> Flux.just((CosmosBulkOperationResponse<TContext>) cosmosBulkOperationResponse));
+            }
+            return Flux.just((CosmosBulkOperationResponse<TContext>) cosmosBulkOperationResponse);
```

Comment:
You could just return Mono , because this is single value on every async value of parent flux.
Flux on flux will be in inefficient imo , please check

Reply:
Added the code to return a Mono